pdg/src/graph.rs

```diff
@@ -119,6 +120,8 @@ pub struct Node {
     pub kind: NodeKind,
     /// The `Node` that produced the input to this operation.
     pub source: Option<NodeId>,
+    /// Additional information generated from the graphs
+    pub node_info: Option<NodeInfo>,
```

Comment:
I'm not sure this needs to be an `Option` as it's never `None` by the end.  Maybe `augment_with_info` should just return a something new where `NodeInfo` is unconditionally set.  I'll think more about a good way to arrange that.